src/data/row.rs

```diff
@@ -45,58 +45,64 @@ impl Row {
             .ok_or_else(|| RowError::ConflictOnEmptyRow.into())
     }
 
-    pub fn new(column_defs: Vec<ColumnDef>, columns: &[Ident], source: &Query) -> Result<Self> {
+    pub fn new(
+        column_defs: Vec<ColumnDef>,
+        columns: &[Ident],
+        source: &Query,
+    ) -> Result<Vec<Self>> {
         let values = match &source.body {
             SetExpr::Values(Values(values)) => values,
             _ => {
                 return Err(RowError::Unreachable.into());
             }
         };
 
-        let values = match values.len() {
-            1 => &values[0],
-            0 => {
-                return Err(RowError::Unreachable.into());
-            }
-            _ => {
-                return Err(RowError::MultiRowInsertNotSupported.into());
-            }
-        };
-
-        column_defs
-            .into_iter()
-            .enumerate()
-            .map(|(i, column_def)| {
-                let ColumnDef {
-                    name,
-                    data_type,
-                    options,
-                    ..
-                } = column_def;
-                let name = name.to_string();
-
-                let i = match columns.len() {
-                    0 => Ok(i),
-                    _ => columns
-                        .iter()
-                        .position(|target| target.value == name)
-                        .ok_or_else(|| RowError::LackOfRequiredColumn(name.clone())),
-                }?;
-
-                let literal = values
-                    .get(i)
-                    .ok_or_else(|| RowError::LackOfRequiredValue(name.clone()))?;
-                let nullable = options
-                    .iter()
-                    .any(|ColumnOptionDef { option, .. }| option == &ColumnOption::Null);
-
-                match literal {
-                    Expr::Value(literal) => Value::from_data_type(data_type, nullable, literal),
-                    Expr::Identifier(Ident { value, .. }) => Ok(Value::Str(value.clone())),
-                    _ => Err(RowError::UnsupportedAstValueType.into()),
-                }
+        if values.is_empty() {
+            return Err(RowError::Unreachable.into());
+        }
+
+        values
+            .iter()
+            .map(|values| {
+                column_defs
+                    .clone()
```

Comment:
The only reason that `Row::new` takes ownership of `column_defs` is that it is fetched by `prepare` function so at least someone should take it.
But at the same time, for the code logic, it's actually enough to use `column_defs` by borrowing.

This `clone()` calls are unnecessary, you will be able to remove this `clone()` calls.

Reply:
I had already tried that already, It does work but then it requires `data_type` on line 96 to be cloned. would that be fine?

Reply:
@ryanhossain9797 Good point, oh. I also saw that `Value::from_data_type`, I think it's better to change `Value::from_data_type` to borrow `data_type` rather than taking ownership.
`Value::from_data_type` is only used here so let's change the type to `&DataType`.

fyi. Though `Value::from_data_type` is used only in `row.rs` but I put this method in `Value` is the case when we think about adding a new value type.
If we consider to add a new value, for example, may be.. `I32`, 32 bit signed integer, at that case if we need to take some effort about where to change, that can be quite tough.
But I wanted to simply add new value type by only modifying `value.rs` file, without thinking about any other files.

Reply:
Yep I can see why it's useful
Anyway, done with changes